File: lib/src/model/ListMultipartUploadsResult.java
package com.alibaba.sdk.android.oss.model;

import android.util.Xml;

import com.alibaba.sdk.android.oss.common.utils.DateUtil;
import com.alibaba.sdk.android.oss.common.utils.OSSUtils;
import com.alibaba.sdk.android.oss.internal.ResponseMessage;

import org.xmlpull.v1.XmlPullParser;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by jingdan on 2018/2/13.
 */

 class ListMultipartUploadsResult extends OSSResult {

     String bucketName;

     String keyMarker;

     String delimiter;

     String prefix;

     String uploadIdMarker;

     int maxUploads;

     bool isTruncated;

     String nextKeyMarker;

     String nextUploadIdMarker;

     List<MultipartUpload> multipartUploads = new ArrayList<MultipartUpload>();

     List<String> commonPrefixes = new ArrayList<String>();

     String getBucketName() {
        return bucketName;
    }

     void setBucketName(String bucketName) {
        this.bucketName = bucketName;
    }

     String getKeyMarker() {
        return keyMarker;
    }

     void setKeyMarker(String keyMarker) {
        this.keyMarker = keyMarker;
    }

     String getUploadIdMarker() {
        return uploadIdMarker;
    }

     void setUploadIdMarker(String uploadIdMarker) {
        this.uploadIdMarker = uploadIdMarker;
    }

     String getNextKeyMarker() {
        return nextKeyMarker;
    }

     void setNextKeyMarker(String nextKeyMarker) {
        this.nextKeyMarker = nextKeyMarker;
    }

     String getNextUploadIdMarker() {
        return nextUploadIdMarker;
    }

     void setNextUploadIdMarker(String nextUploadIdMarker) {
        this.nextUploadIdMarker = nextUploadIdMarker;
    }

     int getMaxUploads() {
        return maxUploads;
    }

     void setMaxUploads(int maxUploads) {
        this.maxUploads = maxUploads;
    }

     bool isTruncated() {
        return isTruncated;
    }

     void setTruncated(bool isTruncated) {
        this.isTruncated = isTruncated;
    }

     List<MultipartUpload> getMultipartUploads() {
        return multipartUploads;
    }

     void setMultipartUploads(List<MultipartUpload> multipartUploads) {
        this.multipartUploads.clear();
        if (multipartUploads != null && !multipartUploads.isEmpty()) {
            this.multipartUploads.addAll(multipartUploads);
        }
    }

     void addMultipartUpload(MultipartUpload multipartUpload) {
        this.multipartUploads.add(multipartUpload);
    }

     String getDelimiter() {
        return delimiter;
    }

     void setDelimiter(String delimiter) {
        this.delimiter = delimiter;
    }

     String getPrefix() {
        return prefix;
    }

     void setPrefix(String prefix) {
        this.prefix = prefix;
    }

     List<String> getCommonPrefixes() {
        return commonPrefixes;
    }

     void setCommonPrefixes(List<String> commonPrefixes) {
        this.commonPrefixes.clear();
        if (commonPrefixes != null && !commonPrefixes.isEmpty()) {
            this.commonPrefixes.addAll(commonPrefixes);
        }
    }

     void addCommonPrefix(String commonPrefix) {
        this.commonPrefixes.add(commonPrefix);
    }

     ListMultipartUploadsResult parseData(ResponseMessage responseMessage) throws Exception {
        List<MultipartUpload> uploadList = new ArrayList<MultipartUpload>();
        MultipartUpload upload = null;
        bool isCommonPrefixes = false;
        XmlPullParser parser = Xml.newPullParser();
        parser.setInput(responseMessage.getContent(), "utf-8");
        int eventType = parser.getEventType();
        while (eventType != XmlPullParser.END_DOCUMENT) {
            switch (eventType) {
                case XmlPullParser.START_TAG:
                    String name = parser.getName();
                    if ("Bucket".equals(name)) {
                        setBucketName(parser.nextText());
                    } else if ("Delimiter".equals(name)) {
                        setDelimiter(parser.nextText());
                    } else if ("Prefix".equals(name)) {
                        if (isCommonPrefixes) {
                            String commonPrefix = parser.nextText();
                            if (!OSSUtils.isEmptyString(commonPrefix)) {
                                addCommonPrefix(commonPrefix);
                            }
                        } else {
                            setPrefix(parser.nextText());
                        }
                    } else if ("MaxUploads".equals(name)) {
                        String maxUploads = parser.nextText();
                        if (!OSSUtils.isEmptyString(maxUploads)) {
                            setMaxUploads(Integer.valueOf(maxUploads));
                        }
                    } else if ("IsTruncated".equals(name)) {
                        String isTruncated = parser.nextText();
                        if (!OSSUtils.isEmptyString(isTruncated)) {
                            setTruncated(bool.valueOf(isTruncated));
                        }
                    } else if ("KeyMarker".equals(name)) {
                        setKeyMarker(parser.nextText());
                    } else if ("UploadIdMarker".equals(name)) {
                        setUploadIdMarker(parser.nextText());
                    } else if ("NextKeyMarker".equals(name)) {
                        setNextKeyMarker(parser.nextText());
                    } else if ("NextUploadIdMarker".equals(name)) {
                        setNextUploadIdMarker(parser.nextText());
                    } else if ("Upload".equals(name)) {
                        upload = new MultipartUpload();
                    } else if ("Key".equals(name)) {
                        upload.setKey(parser.nextText());
                    } else if ("UploadId".equals(name)) {
                        upload.setUploadId(parser.nextText());
                    } else if ("Initiated".equals(name)) {
                        upload.setInitiated(DateUtil.parseIso8601Date(parser.nextText()));
                    } else if ("StorageClass".equals(name)) {
                        upload.setStorageClass(parser.nextText());
                    } else if ("CommonPrefixes".equals(name)) {
                        isCommonPrefixes = true;
                    }
                    break;
                case XmlPullParser.END_TAG:
                    if ("Upload".equals(parser.getName())) {
                        uploadList.add(upload);
                    } else if ("CommonPrefixes".equals(parser.getName())) {
                        isCommonPrefixes = false;
                    }
                    break;
            }

            eventType = parser.next();
            if (eventType == XmlPullParser.TEXT) {
                eventType = parser.next();
            }
        }

        if (uploadList.size() > 0) {
            setMultipartUploads(uploadList);
        }

        return this;
    }
}

File: lib/src/model/list_multipart_uploads_result.dart
import 'package:aliyun_oss_dart_sdk/src/common/lib_common.dart';
import 'package:aliyun_oss_dart_sdk/src/internal/response_message.dart';
import 'multipart_upload.dart';
import 'oss_result.dart';

class ListMultipartUploadsResult extends OSSResult {
  String? bucketName;

  String? keyMarker;

  String? delimiter;

  String? prefix;

  String? uploadIdMarker;

  int maxUploads = 0;

  bool isTruncated = false;

  String? nextKeyMarker;

  String? nextUploadIdMarker;

  final List<MultipartUpload> _multipartUploads = [];

  final List<String> _commonPrefixes = [];

  List<MultipartUpload> getMultipartUploads() {
    return _multipartUploads;
  }

  void setMultipartUploads(List<MultipartUpload>? multipartUploads) {
    _multipartUploads
      ..clear()
      ..addAll(multipartUploads ?? []);
  }

  void addMultipartUpload(MultipartUpload multipartUpload) {
    _multipartUploads.add(multipartUpload);
  }

  List<String> getCommonPrefixes() {
    return _commonPrefixes;
  }

  void setCommonPrefixes(List<String>? commonPrefixes) {
    _commonPrefixes
      ..clear()
      ..addAll(commonPrefixes ?? []);
  }

  void addCommonPrefix(String commonPrefix) {
    _commonPrefixes.add(commonPrefix);
  }

  ListMultipartUploadsResult parseData(ResponseMessage responseMessage) {
    List<MultipartUpload> uploadList = [];
    MultipartUpload? upload;
    bool isCommonPrefixes = false;
    XmlPullParser parser = Xml.newPullParser();
    parser.setInput(responseMessage.content, "utf-8");
    int eventType = parser.getEventType();
    while (eventType != XmlPullParser.END_DOCUMENT) {
      switch (eventType) {
        case XmlPullParser.START_TAG:
          String name = parser.getName();
          if ("Bucket" == name) {
            bucketName = parser.nextText();
          } else if ("Delimiter" == name) {
            delimiter = parser.nextText();
          } else if ("Prefix" == name) {
            if (isCommonPrefixes) {
              String commonPrefix = parser.nextText();
              if (commonPrefix.notNullOrEmpty) {
                addCommonPrefix(commonPrefix);
              }
            } else {
              prefix = parser.nextText();
            }
          } else if ("MaxUploads" == name) {
            String? maxUploadStr = parser.nextText();
            if (maxUploadStr.notNullOrEmpty) {
              maxUploads = int.parse(maxUploadStr!);
            }
          } else if ("IsTruncated" == name) {
            String? truncated = parser.nextText();
            if (truncated.notNullOrEmpty) {
              isTruncated = truncated!.equalsIgnoreCase('true');
            }
          } else if ("KeyMarker" == name) {
            keyMarker = parser.nextText();
          } else if ("UploadIdMarker" == name) {
            uploadIdMarker = parser.nextText();
          } else if ("NextKeyMarker" == name) {
            nextKeyMarker = parser.nextText();
          } else if ("NextUploadIdMarker" == name) {
            nextUploadIdMarker = parser.nextText();
          } else if ("Upload" == name) {
            upload = MultipartUpload();
          } else if ("Key" == name) {
            upload?.key = parser.nextText();
          } else if ("UploadId" == name) {
            upload?.uploadId = parser.nextText();
          } else if ("Initiated" == name) {
            upload?.initiated = DateUtil.parseIso8601Date(parser.nextText());
          } else if ("StorageClass" == name) {
            upload?.storageClass = parser.nextText();
          } else if ("CommonPrefixes" == name) {
            isCommonPrefixes = true;
          }
          break;
        case XmlPullParser.END_TAG:
          if ("Upload" == parser.getName()) {
            uploadList.add(upload);
          } else if ("CommonPrefixes" == parser.getName()) {
            isCommonPrefixes = false;
          }
          break;
      }

      eventType = parser.next();
      if (eventType == XmlPullParser.TEXT) {
        eventType = parser.next();
      }
    }

    if (uploadList.isNotEmpty) {
      setMultipartUploads(uploadList);
    }

    return this;
  }
}
